# Ecommerce Sales Dashboard

## Notebook 01 – Data Understanding

### Author
Divya Sesilia

### Project
End-to-End Data Analytics Project using SQL, Python, Excel, and Power BI

### Dataset

Superstore Sales Dataset

### Tools Used

- Python
- Pandas
- Jupyter Notebook

### Notebook Version

1.0

### Objective

The objective of this notebook is to understand the dataset before performing any data cleaning or analysis.

### Tasks

- Load the dataset
- Explore the dataset structure
- Check rows and columns
- Identify data types
- Find missing values
- Find duplicate records
- Generate descriptive statistics
- Record observations

## Load Required Library

Pandas is used to load, manipulate, and analyze tabular datasets.

In [1]:
import pandas as pd

## Load Dataset

Load the Superstore dataset into a Pandas DataFrame.

In [2]:
df = pd.read_csv("../data/raw/superstore.csv")


In [3]:
# Display the first five records
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,8/11/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,8/11/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,12/6/2016,16/6/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,11/10/2015,18/10/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,11/10/2015,18/10/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [4]:
rows, columns = df.shape

print(f"Number of Rows    : {rows}")
print(f"Number of Columns : {columns}")

Number of Rows    : 9994
Number of Columns : 21


In [5]:
print("Column Names:\n")

for col in df.columns:
    print(col)

Column Names:

Row ID
Order ID
Order Date
Ship Date
Ship Mode
Customer ID
Customer Name
Segment
Country
City
State
Postal Code
Region
Product ID
Category
Sub-Category
Product Name
Sales
Quantity
Discount
Profit


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

## Initial Observations

- The dataset contains **9,994 records** and **21 columns**.
- There are **no missing values** in any column.
- The dataset includes customer, product, sales, shipping, and regional information.
- `Order Date` and `Ship Date` are stored as text (`object`) and should be converted to datetime.
- `Sales`, `Profit`, and `Discount` are numerical columns that will be used for business analysis.
- The dataset is clean enough to begin exploratory analysis after converting the date columns.

In [7]:
# Generate descriptive statistics for numerical columns
df.describe()

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,4997.500000,55190.379428,229.858001,3.789574,0.156203,28.656896
std,2885.163629,32063.693350,623.245101,2.225110,0.206452,234.260108
min,1.000000,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,56430.500000,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,90008.000000,209.940000,5.000000,0.200000,29.364000
max,9994.000000,99301.000000,22638.480000,14.000000,0.800000,8399.976000


## Statistical Summary

### Observations

- Average Sales per Order: 229.86
- Average Profit per Order: 28.66
- Maximum Discount Offered: 80%
- Some orders incur significant losses.
- Sales values vary widely, indicating the presence of high-value orders.

In [8]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

## Missing Values Analysis

There are no missing values in the dataset.

The dataset is complete and ready for further analysis.

In [9]:
df.duplicated().sum()

np.int64(0)

## Duplicate Records Analysis

No duplicate records were found.

The dataset does not require duplicate removal.

In [10]:
df.sample(5)

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
9231,9232,CA-2014-148383,27/12/2014,31/12/2014,Standard Class,RP-19390,Resi Pölking,Consumer,United States,Phoenix,...,85023,West,OFF-BI-10003650,Office Supplies,Binders,GBC DocuBind 300 Electric Binding Machine,946.764,6,0.70,-694.2936
9049,9050,CA-2014-124513,21/9/2014,25/9/2014,Standard Class,SL-20155,Sara Luxemburg,Home Office,United States,New York City,...,10011,East,OFF-AR-10002956,Office Supplies,Art,Boston 16801 Nautilus Battery Pencil Sharpener,66.030,3,0.00,17.1678
9839,9840,US-2016-125402,25/9/2016,1/10/2016,Standard Class,DL-12865,Dan Lawera,Consumer,United States,Long Beach,...,90805,West,TEC-CO-10001943,Technology,Copiers,Canon PC-428 Personal Copier,479.976,3,0.20,161.9919
4023,4024,CA-2017-130764,27/10/2017,28/10/2017,First Class,JO-15145,Jack O'Briant,Corporate,United States,San Francisco,...,94110,West,FUR-BO-10003034,Furniture,Bookcases,"O'Sullivan Elevations Bookcase, Cherry Finish",556.665,5,0.15,6.5490
9704,9705,CA-2017-147032,31/7/2017,4/8/2017,Standard Class,LB-16795,Laurel Beltran,Home Office,United States,Wilmington,...,19805,East,OFF-BI-10002897,Office Supplies,Binders,"Black Avery Memo-Size 3-Ring Binder, 5 1/2"" x ...",11.010,3,0.00,5.3949


## Random Sample Review

Randomly inspecting records helps verify that the dataset appears consistent and contains realistic business information.

This confirms that the data structure is suitable for further analysis.

# Conclusion

## Summary

- Successfully loaded the Superstore dataset.
- Verified that the dataset contains **9,994 rows** and **21 columns**.
- Confirmed there are **no missing values**.
- Confirmed there are **no duplicate records**.
- Identified that **Order Date** and **Ship Date** need conversion to datetime.
- Identified the key business metrics: Sales, Profit, Quantity, and Discount.

## Next Steps

In the next notebook, we will:

- Convert date columns to datetime.
- Validate data types.
- Prepare a cleaned dataset.
- Export the cleaned dataset for SQL, Excel, and Power BI analysis.